crates/sui-storage/src/event_store/sql.rs

```diff
@@ -495,7 +576,55 @@ mod tests {
         Ok(())
     }
 
-    // TODO: test MoveEvents
+    // Test for reads by move event
+    #[tokio::test]
+    async fn test_eventstore_move_events() -> Result<(), EventStoreError> {
+        telemetry_subscribers::init_for_testing();
+
+        // Initialize store
+        let db = SqlEventStore::new_sqlite(":memory:").await?;
+        db.initialize().await?;
+
+        // Insert some records
+        info!("Inserting records!");
+        let to_insert = test_events();
+        db.add_events(&to_insert, 1).await?;
+        info!("Done inserting");
+
+        // Query for the Move event and validate basic fields
+        let mut event_it = db
+            .events_for_transaction(to_insert[5].tx_digest.unwrap())
+            .await?;
+        let move_event = event_it.next().expect("No move events in result!!");
+        assert_eq!(event_it.next(), None); // Should be no more events, just that one
+
+        test_queried_event_vs_test_envelope(&move_event, &to_insert[5]);
+        assert_eq!(
+            move_event.function_name,
+            to_insert[5].event.function_name().map(SharedStr::from)
+        );
+        assert_eq!(move_event.fields.len(), 2);
+
+        // Query by module ID
+        let mod_id = ModuleId::new(
+            *to_insert[5].event.package_id().unwrap(),
+            Identifier::new(to_insert[5].event.module_name().unwrap()).unwrap(),
+        );
+        let event_it = db
+            .events_by_module_id(1_000_000, 1_005_001, mod_id, 2)
+            .await?;
+        let queried_events: Vec<_> = event_it.collect();
+        assert_eq!(queried_events.len(), 1);
+
+        test_queried_event_vs_test_envelope(&queried_events[0], &to_insert[5]);
+        assert_eq!(
+            queried_events[0].function_name,
+            to_insert[5].event.function_name().map(SharedStr::from)
+        );
+        assert_eq!(queried_events[0].fields.len(), 2);
+
+        Ok(())
```

Comment:
let's add test cases for:
1. having a big limit and getting all eligible results
2. having a smaller limit than eligible results

Reply:
Both are tested already.  1. is tested in `test_eventstore_basic_insert_read` (limit=20, checks that all 6 events returned).   2. is tested in `test_by_type()` with limit of 1